sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReactorReceiver.java

```diff
@@ -346,6 +346,10 @@ private void updateOutcome(String lockToken, Delivery delivery) {
     }
 
     private void cleanupWorkItems() {
+        if (pendingUpdates.isEmpty()) {
+            return;
+        }
```

Comment:
Awesome! this will help to cut down the below excessive verbose logging (that was a pain when we analyze the logs)